sky/utils/volume.py

```diff
@@ -59,14 +65,38 @@ def __init__(self,
         self.volume_config: models.VolumeConfig = volume_config
         self.is_ephemeral: bool = is_ephemeral
 
-    def pre_mount(self) -> None:
-        """Update the volume status before actual mounting."""
+    def pre_mount(self, cluster_name: Optional[str] = None) -> None:
+        """Update the volume status before actual mounting.
+
+        Args:
+            cluster_name: Name of the cluster this volume is being mounted to.
+                         Used to show warnings about existing attachments.
+        """
         # Skip pre_mount for ephemeral volumes as they don't exist yet
         if self.is_ephemeral:
             return
         # TODO(aylei): for ReadWriteOnce volume, we also need to queue the
         # mount request if the target volume is already mounted to another
         # cluster. For now, we only support ReadWriteMany volume.
+
+        # Check if volume is already attached to a cluster and warn if
+        # relaunching
+        if cluster_name is not None:
+            try:
+                # Get the list of clusters currently using this volume
+                _, usedby_clusters = provision.get_volume_usedby(
+                    self.volume_config.cloud, self.volume_config)
+
+                if usedby_clusters and cluster_name in usedby_clusters:
+                    logger.info(
+                        f'{colorama.Fore.YELLOW}Volume {self.volume_name!r} is '
+                        f'already mounted to cluster {cluster_name!r}. '
+                        f'Relaunching with the same volume.'
+                        f'{colorama.Style.RESET_ALL}')
+            except Exception as e:  # pylint: disable=broad-except
+                # Don't fail if we can't check volume usage, just log debug
+                logger.debug(f'Failed to check volume usage for warning: {e}', exc_info=True)
```

Comment:
The purpose of the warning is to tell a user if they specify a new volume that is not mounted to the existing cluster in the SkyPilot YAML, the new volume will not be mounted to the cluster. (similar to the handling like this: https://github.com/skypilot-org/skypilot/pull/6912)